File: genPrimes.js
/* generates an array of prime numbers up to
  ( <= to ) the argument, then returns that array */
exports.primes = (upTo) => {
  let found = [2,3]
  for (let i = 5; i <= upTo; i += 2 ) {
    let r = Math.ceil(Math.sqrt(i))
    for (let j = 1; j < found.length; j++ ) {
      if (i % found[j] === 0) {
        break
      } else if (found[j] >= r) {
        found.push(i)
        break
      }
    }
  }
  return found
}

File: genPrimes.py
import math


# not very efficient, should improve  :)
def genUpTo(x):
    found = [2,3]
    i = 5
    while i <= x:
        r = math.ceil(math.sqrt(i))
        j = 1
        while j <= len(found):
            if i % found[j] == 0:
                break
            if found[j] < r:
                j += 1
                continue
            found.append(i)
            break
        i += 2 
    return found

# print(genUpTo(100000))